File: src/main/java/com/sonner/api/services/ClienteService.java
package com.sonner.api.services;

import com.sonner.api.entidades.Cliente;
import com.sonner.api.repositorios.ClienteRepository;
import com.sonner.api.services.base.AbstractService;
import org.springframework.stereotype.Service;

@Service
public class ClienteService extends AbstractService<Cliente, Integer, ClienteRepository> {

    public ClienteService(ClienteRepository repository) {
        super(repository);
    }

    public Cliente getByNome(String nome) {

        return repository.findByNomeClienteLike(nome+"%").orElse(null);
    }

}

File: web-angular/front/src/app/services/cliente.service.ts
import {AbstractService} from "./base/abstract.service";
import {Cliente} from "../model/cliente";
import {HttpClient} from "@angular/common/http";
import {Injectable} from "@angular/core";
import {Observable} from "rxjs";
import {API_CONFIG} from "../configs/api.config";


@Injectable({
  providedIn:'root'
})
export class ClienteService extends AbstractService<Cliente>{

  constructor(protected override http: HttpClient) {
    super(http, 'cliente');
  }

  findByName(nome:string): Observable<Array<Cliente>>{
    return  this.http.get<Array<Cliente>>(`${API_CONFIG.baseUrl}/${this.url}/nome/${nome}`);

  }
}
